### PR TITLE
[FIX] Move invoice's attachment name to noupdate record

### PR DESCRIPTION
**Pull request by hurrinico**
Description of the issue/feature this PR addresses:
Name of invoice's attachment is reset after updating invoice's module, cause attachment_use and attachment tag are not in a noupdate tag
Current behavior before PR:
Invoice's attachment name is reset after updating module
Desired behavior after PR is merged:
Invoice's attachment name is not reset after updating module
## 

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by eLBati** [comment]
@hurrinico any news about the CLA?

**Comment by hurrinico** [comment]
@eLBati i've added hurrinico.md

**Comment by dreispt** [comment]
@odony Food for thought:
It could make sense to move the `noupdate` attribute to the `<record>` tag (and equivalents), and remove completely the `<data>` element...

**Comment by odony** [comment]
@dreispt supporting ` @noupdate` on `<record>` is a good idea, we just have to carefully define the semantics (+ tests) of nested cases with `<data>`. I'm not sure we should support this on all equivalent elements, though, as they do not have the required granularity to be really useful (cfr the very PR we're discussing on ;-))
However I'd suggest keeping `<data>`, not just for compatibility but also because it lets you factor out the `noupdate` flag, and organize your data section in meaningful ways. It also serves as a dummy "root"  element in inherited views with multiple inheritance sections, in order to have a well-formed XML doc.

If anyone would like to make a proof-of-concept PR for this (for master), I'd be happy to review/discuss.

By the way, IIRC the root element of XML files can be a choice of `<odoo>` / `<openerp>` / `<data>`, so we could simplify/generalize our files as
```
        <data>
           <record noupdate="1"/>
           <record/>

           <data noupdate="1">
               <record/>
               ...
           </data>
        </data>
```

But we're quite off-topic ;-)

File: addons/account/account_report.xml [hunk 1]
```diff
@@ -165,4 +163,10 @@
...
         />
     </data>
-</openerp>
+    <data noupdate="1">
+        <record id="account.account_invoices" model="ir.actions.report.xml">
+            <field name="attachment_use">True</field>
+            <field name="attachment">(object.state in ('open','paid')) and ('INV'+(object.number or '').replace('/','')+'.pdf')</field>
+        </record>
+    </data>
+</openerp>
```
Comment (eLBati) [inline addons/account/account_report.xml]: @hurrinico please add newline

Comment (hurrinico) [inline addons/account/account_report.xml]: @eLBati added

**Comment by eLBati** [comment]
:+1:

**Comment by tafaRU** [comment]
Overseeded by https://github.com/odoo/odoo/pull/15522

**Comment by tafaRU** [comment]
@hurrinico, please close it!

**Pull request closed without merging**